plugins/module_utils/network/ios/argspec/snmp_server/snmp_server.py

```diff
@@ -218,26 +255,37 @@ class Snmp_serverArgs(object):  # pylint: disable=R0903
                         },
                         "eigrp": {"type": "bool"},
                         "entity": {"type": "bool"},
+                        "entity_diag": {
+                            "options": {
+                                "boot_up_fail": {"type": "bool"},
+                                "enable": {"type": "bool"},
+                                "hm_test_recover": {"type": "bool"},
+                                "hm_thresh_reached": {"type": "bool"},
+                                "scheduled_test_fail": {"type": "bool"},
+                            },
+                            "type": "dict",
+                        },
+                        "entity_perf": {
+                            "options": {
+                                "enable": {"type": "bool"},
+                                "throughput_notif": {"type": "bool"},
+                            },
+                            "type": "dict",
+                        },
+                        "entity_state": {"type": "bool"},
                         "energywise": {"type": "bool"},
                         "envmon": {
                             "options": {
-                                "fan": {
```

Comment:
@earendilfr The fan option here has changed from type dict to bool, this also is considered as a deprecation since lot of people might be using this attribute in their playbooks, hence we cant remove this immediately.

 We must follow similar deprecation process for this and all the changed attributes even if it a minor change.

Reply:
@roverflow : as I change the type, did I need to use keep the `fan` parameter as dict to keep the compatibility and create a new key as `fan_enable` (or something like that) for the new one ?
Or just a warning notice in documentation is enough ?

Reply:
I have take a look and only this one and the parameter mpls_vpn has changed.
All the others have the same value

Reply:
@earendilfr Yes!!, you have to keep the `fan` parameter and create a new key as you mentioned `fan_enable` and in docs you would have to specify that the `fan` parameter is deprecated. As always thank you for taking your time to do this contribution 😄 .